commit d4b4e17d2c70c8d498ad33422cf847d659b5b0cf
Author: Evgeniy Stepanov <eugeni.stepanov@gmail.com>
Date:   Thu Feb 28 01:54:55 2019 +0000

    [sanitizer] Got rid of text relocations in i386 vfork interceptor.
    
    llvm-svn: 355064

diff --git a/compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors_vfork_i386.inc.S b/compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors_vfork_i386.inc.S
index f130b7250ce..d693aa3e050 100644
--- a/compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors_vfork_i386.inc.S
+++ b/compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors_vfork_i386.inc.S
@@ -13,8 +13,12 @@ ASM_WRAPPER_NAME(vfork):
         mov     %ecx, (%eax)
         add     $16, %esp
 
-
-        call    *_ZN14__interception10real_vforkE
+        call    .L0$pb
+.L0$pb:
+        pop     %eax
+.Ltmp0:
+        add     $_GLOBAL_OFFSET_TABLE_+(.Ltmp0-.L0$pb), %eax
+        call    *_ZN14__interception10real_vforkE@GOTOFF(%eax)
 
         // Restore return address from the spill area.
         sub     $16, %esp
